substrate/frame/election-provider-multi-block/src/signed/tests.rs

```diff
@@ -633,4 +640,626 @@ mod e2e {
+					crate::verifier::Event::Verified(1, 0),
+					crate::verifier::Event::Verified(0, 0),
+					crate::verifier::Event::VerificationFailed(0, FeasibilityError::InvalidScore),
+				]
+			);
+
+			// Check verifier status - should still be ongoing.
+			assert!(matches!(VerifierPallet::status(), crate::verifier::Status::Ongoing(_)));
+
+			roll_next(); // Block 4: processes page 2 of good solution
+			roll_next(); // Block 5: processes page 1 of good solution
+			roll_next(); // Block 6: processes page 0 of good solution
+
+			// Check verifier status - should be Nothing after completion
+			assert_eq!(VerifierPallet::status(), crate::verifier::Status::Nothing);
+
+			// Check good solution is fully verified and removed from queue
+			assert_eq!(Submissions::<Runtime>::submitters_count(current_round), 0);
+			assert!(!Submissions::<Runtime>::has_leader(current_round));
+		});
+	}
+
+	#[test]
+	fn after_rejecting_does_not_call_verifier_start_if_no_leader_exists() {
+		ExtBuilder::signed().build_and_execute(|| {
+			roll_to_signed_open();
+			assert_full_snapshot();
+
+			// Submit only an invalid solution
+			let score = ElectionScore { minimal_stake: 10, sum_stake: 10, sum_stake_squared: 100 };
+			assert_ok!(SignedPallet::register(RuntimeOrigin::signed(99), score));
+			assert_ok!(SignedPallet::submit_page(
+				RuntimeOrigin::signed(99),
+				0,
+				Some(Default::default())
+			));
+
+			// Verify we have exactly one submission
+			let current_round = SignedPallet::current_round();
+			assert_eq!(Submissions::<Runtime>::submitters_count(current_round), 1);
+			assert!(Submissions::<Runtime>::has_leader(current_round));
+
+			roll_to_signed_validation_open();
+
+			assert!(matches!(VerifierPallet::status(), crate::verifier::Status::Ongoing(_)));
+
+			roll_next(); // Block 1: processes page 2 (empty)
+			roll_next(); // Block 2: processes page 1  (empty)
+			roll_next(); // Block 3: processes page 0 (the only submitted page)
+
+			// After 3 blocks, the invalid solution should be processed and discarded
+			// Verify no-restart conditions are met
+			assert!(crate::Pallet::<Runtime>::current_phase().is_signed_validation());
+			assert!(!Submissions::<Runtime>::has_leader(current_round));
+			assert_eq!(Submissions::<Runtime>::submitters_count(current_round), 0);
+			assert_eq!(VerifierPallet::status(), crate::verifier::Status::Nothing);
+
+			// Check that expected events were emitted for the rejection
+			assert_eq!(
+				signed_events(),
+				vec![
+					Event::Registered(
+						0,
+						99,
+						ElectionScore { minimal_stake: 10, sum_stake: 10, sum_stake_squared: 100 }
+					),
+					Event::Stored(0, 99, 0),
+					Event::Slashed(0, 99, 6),
+				]
+			);
+			assert_eq!(
+				verifier_events(),
+				vec![
+					crate::verifier::Event::Verified(2, 0),
+					crate::verifier::Event::Verified(1, 0),
+					crate::verifier::Event::Verified(0, 0),
+					crate::verifier::Event::VerificationFailed(0, FeasibilityError::InvalidScore),
+				]
+			);
```

Comment:
To actually test what was intended to be tested here, you should now progress at least one block, and ensure `::start` signal is no longer set ;)